src/lib/components/Switch/root.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts" context="module">
+	import type { BaseProps } from '$lib/types';
+	import { uniqueContext } from '$lib/helpers/uniqueContext';
+
+	export type SwitchProps = BaseProps<HTMLButtonElement> & {
+		checked?: boolean;
+		defaultChecked?: boolean;
+		required?: boolean;
+		disabled?: boolean;
+		name?: string;
+		value?: string;
+	};
+
+	type SwitchContext = {
+		checked: Writable<SwitchProps['checked']>;
+		disabled: Readable<SwitchProps['disabled']>;
+	};
+
+	const { getContext, setContext } = uniqueContext<SwitchContext>();
+	export const getSwitchContext = getContext;
+
+	export function getState(checked: boolean | undefined) {
+		return checked ? 'checked' : 'unchecked';
+	}
+</script>
+
+<script lang="ts">
+	type $$Props = SwitchProps;
+
+	import { derived, writable, type Readable, type Writable } from 'svelte/store';
+	import { controllableState } from '$lib/helpers/controllableState';
+
+	export let defaultChecked: $$Props['checked'] = false;
+	export let required: $$Props['required'] = false;
+	export let value: $$Props['value'] = 'on';
+
+	export let checked: $$Props['checked'] = defaultChecked;
+	const writableChecked = controllableState(checked, (v) => (checked = v));
+	$: $writableChecked = checked;
+
+	export let disabled: $$Props['disabled'] = false;
+	const writableDisabled = writable(disabled);
+	$: if (typeof disabled !== 'undefined') $writableDisabled = disabled;
+
+	let button: HTMLButtonElement;
+	$: isFormControl = button ? button.closest('form') : true;
+
+	setContext({
+		checked: writableChecked,
+		disabled: derived(writableDisabled, (v) => v)
+	});
+</script>
+
+<button
+	bind:this={button}
+	type="button"
+	role="switch"
+	class={$$props.class}
+	aria-checked={checked}
+	aria-required={required}
+	aria-labelledby={$$props.id}
```

Comment:
Is this right? Shouldn't it be the label of the `<label>` element (which this component wont know about)

Reply:
You're absolutely right, I'll remove this. 

I tried to get the switch to pass accessibility guidelines, that require buttons to have descriptive text on them, but I don't know if that's possible with the Radix's structure. Even Radix itself doesn't adhere to it.

Reply:
Yeah if you see my test on my pr, i add aria-label in the test. Thats also how Radix does it.